rules/CodeQuality/Rector/Attribute/OrderByKeyToClassConstRector.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Doctrine\CodeQuality\Rector\Attribute;
+
+use Doctrine\Common\Collections\Criteria;
+use PhpParser\Node;
+use PhpParser\Node\Attribute;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Scalar\String_;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Doctrine\Tests\CodeQuality\Rector\Attribute\OrderByKeyToClassConstRector\OrderByKeyToClassConstRectorTest
+ */
+final class OrderByKeyToClassConstRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Replace OrderBy Attribute ASC/DESC with class constant from Criteria',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+use Doctrine\ORM\Mapping as ORM;
+
+class ReplaceOrderByAscWithClassConstant
+{
+    #[ORM\OrderBy(['createdAt' => 'ASC'])]
+    protected \DateTimeInterface $messages;
+}
+?>
+CODE_SAMPLE
+
+                    ,
+                    <<<'CODE_SAMPLE'
+use Doctrine\ORM\Mapping as ORM;
+
+class ReplaceOrderByAscWithClassConstant
+{
+    #[ORM\OrderBy(['createdAt' => \Doctrine\Common\Collections\Criteria::ASC])]
+    protected \DateTimeInterface $messages;
+}
+?>
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Attribute::class];
+    }
+
+    /**
+     * @param Attribute $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        // If Attribute is not OrderBy, return null
+        if ($node->name->toString() !== 'Doctrine\ORM\Mapping\OrderBy') {
+            return null;
+        }
+
+        if (!$node->args[0]->value instanceof Array_){
+            return null;
+        }
+
+        if (!$node->args[0]->value->items[0] instanceof Node\Expr\ArrayItem) {
+            return null;
+        }
+        if (!$node->args[0]->value->items[0]->value instanceof String_) {
+            return null;
+        }
+
+        // If Attribute value from key is not `ASC` or `DESC`, return null
+        if (! in_array($node->args[0]->value->items[0]->value->value, ['ASC', 'asc', 'DESC', 'desc'])) {
+            return null;
+        }
+
+        $upper = mb_strtoupper($node->args[0]->value->items[0]->value->value);
```

Comment:
there is no unicode in the above list string, so normal `strtoupper` should fine.